### PR TITLE
move libuv callbacks to julia

### PR DESCRIPTION
**Pull request by JeffBezanson**
This simplifies the code and avoids some extra argument boxing and dynamic dispatching.

The `close` callback seems to be the difficult one.

@Keno @vtjnash What is the `isopen` callback? It doesn't appear to be used; is that right?

### CODE REVIEW

**Comment by vtjnash** [comment]
it is (was?) used by `jl_uv_close` to check the state of the julia side of the object (to prevent double-close, and other such race conditions during the atexit hook)

**Comment by vtjnash** [comment]
:100:

**Comment by JeffBezanson** [comment]
It really doesn't look like it's used anymore. Maybe it should be?

**Comment by Keno** [comment]
No, I believe it might have been removed when I fixed the error conditions for double-close on the libuv side.

**Comment by JeffBezanson** [comment]
Cool. So that leaves the close callback. Given how it's used in the shutdown process, is it even worth moving to julia? We could special case just that one; very easy since it only takes one argument.

**Comment by JeffBezanson** [comment]
Ah, I also see now that `jl_callback_call` has a built in check for the first argument being NULL. Do we need that? I assumed non-NULL in the julia callbacks; is that valid?

File: src/jl_uv.c [hunk 1]
```diff
@@ -177,94 +81,21 @@ DLLEXPORT void jl_uv_shutdownCallback(uv_shutdown_t *req, int status)
...
-    JULIA_CB(pollcb,handle->data,2,CB_INT32,status,CB_INT32,events);
-}
-
-DLLEXPORT void jl_uv_fspollcb(uv_fs_poll_t *handle, int status, const uv_stat_t *prev, const uv_stat_t *curr)
-{
-    JULIA_CB(fspollcb,handle->data,3,CB_INT32,status,CB_PTR,prev,CB_PTR,curr);
-}
+// getters and setters
+DLLEXPORT void *jl_uv_process_data(uv_process_t *p) { return p->data; }
+DLLEXPORT void *jl_uv_buf_base(const uv_buf_t *buf) { return buf->base; }
```
Comment (Keno) [inline src/jl_uv.c]: We should be able to do this in julia since we have the correct calling convention now.

**Pull request merged**